server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/AvroResolver.java

```diff
@@ -137,22 +127,18 @@ public void initialize(RequestContext requestContext) {
      *
      * @param record list of {@link OneField}
      * @return the constructed {@link OneRow}
-     * @throws Exception if constructing a row from the fields failed
      */
     @Override
-    public OneRow setFields(List<OneField> record) throws Exception {
-        throw new UnsupportedOperationException();
-    }
-
-    /**
-     * Tests if the Avro records are residing inside an AVRO file. If the Avro
-     * records are not residing inside an AVRO file, then they may reside inside
-     * a sequence file, regular file, ...
-     *
-     * @return whether the resource is an Avro file
-     */
-    boolean isAvroFile() {
-        return context.getAccessor().toLowerCase().contains("avro");
+    public OneRow setFields(List<OneField> record) {
+        GenericRecord genericRecord = new GenericData.Record((Schema) context.getMetadata());
+        int cnt = 0;
+        for (OneField field : record) {
+            // Avro does not seem to understand regular byte arrays
+            if (field.val instanceof byte[])
+                field.val = ByteBuffer.wrap((byte[]) field.val);
```

Comment:
we prefer even 1-liners inside if() to be in { } to avoid errors and improve readability.
```suggestion
                {
                    field.val = ByteBuffer.wrap((byte[]) field.val);
                }
```